OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -114,18 +114,19 @@ public FerryUnit(Actor self, Actor cargo)
 			protected override void OnFirstRun(Actor self)
 			{
 				if (!cargo.IsDead)
-					QueueChild(new PickupUnit(self, cargo, 0));
+					QueueChild(new PickupUnit(self, cargo, 0, self.Trait<Carryall>().Info.TargetLineColor));
```

Comment:
You should add a `readonly CarryallInfo carryallInfo;` field and do a single lookup in the constructor.